packages/components/pie-webc-core/src/mixins/formControl/formControlMixin.ts

```diff
@@ -47,11 +48,46 @@ export const FormControlMixin =
                 return this._internals.form;
             }
 
+            // Storing this reference as this.form is not available in disconnectedCallback
+            private _managedForm?: HTMLFormElement;
+
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
             constructor (...args: any[]) {
                 super(...args);
                 this._internals = this.attachInternals();
             }
+
+            connectedCallback (): void {
+                super.connectedCallback();
+
+                if (this.form) {
+                    // Ensure that the form is managed by the PieFormManager
+                    if (!window.pieFormManager) {
+                        window.pieFormManager = new PieFormManager();
+                    }
+
+                    if (window.pieFormManager) {
+                        window.pieFormManager.addForm(this.form);
```

Comment:
a question: have we considered the use-case where two different components are calling the mixin? Do we need to check if the form hasn't been added yet or is there a check somewhere else / not needed?

Reply:
If I understand your question correctly, this line hopefully answers?

https://github.com/justeattakeaway/pie/pull/1374/files#diff-405085fc57869377bc65fdd5d8f99a1b2ae2512d6afbb446713cad440f7466f7R39-R41

Any input is allowed to try to add its form to the manager, but it can only be added once

Reply:
yep exactly nice!